PowerEditor/src/WinControls/DoubleBuffer/DoubleBuffer.cpp

```diff
@@ -0,0 +1,149 @@
+// This file is part of Notepad++ project
+// Copyright (C) 2024 Jiri Hruska <jirka@fud.cz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// at your option any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#include "DoubleBuffer.h"
+#include <CommCtrl.h>
+#include <cassert>
+#include <memory>
+
+void DoubleBuffer::subclass(HWND hWnd)
+{
+	std::unique_ptr<DoubleBuffer> self(new DoubleBuffer);
+	if (::SetWindowSubclass(hWnd, s_subclassWndProc, 0, reinterpret_cast<DWORD_PTR>(self.get())))
+	{
+		self.release();
```

Comment:
👍  Clever use of the unique_ptr.

If the subclassion succeeds, the released unmanaged memory (new DoubleBuffer object) will be destroyed at the DoubleBuffer::subclassWndProc -> WM_NCDESTROY time later.